alf/networks/relu_mlp_test.py

```diff
@@ -118,22 +116,77 @@ def test_compute_vjp(self, hidden_layers=(2, ), batch_size=2,
         approach is consistent with the one computed by calling autograd.
         """
         output_size = 4
+        partial_idx = [0, 2]
         spec = TensorSpec((input_size, ))
         mlp = ReluMLP(
             spec, output_size=output_size, hidden_layers=hidden_layers)
 
-        # compute vjp using direct approach
+        # compute vjp and partial using direct approach
         x = torch.randn(batch_size, input_size, requires_grad=True)
         vec = torch.randn(batch_size, output_size)
         x1 = x.detach().clone()
-        x1.requires_grad = True
         vjp, _ = mlp.compute_vjp(x1, vec)
+        vjp_partial, _ = mlp.compute_vjp(
+            x1, vec, output_partial_idx=partial_idx)
 
-        # # compute jac using autograd
+        # # compute vjp using autograd
         y, _ = mlp(x)
         vjp2 = torch.autograd.grad(y, x, grad_outputs=vec)[0]
 
+        # # compute partial vjp using autograd
+        x2 = x.detach().clone()
+        x2.requires_grad = True
+        y2, _ = mlp(x2)
+        mask_idx = np.setdiff1d(range(vec.shape[1]), partial_idx)
+        vec[:, mask_idx] = 0.
+        vjp2_partial = torch.autograd.grad(y2, x2, grad_outputs=vec)[0]
+
         self.assertArrayEqual(vjp, vjp2, 1e-6)
+        self.assertArrayEqual(vjp_partial, vjp2_partial, 1e-6)
+
+    @parameterized.parameters(
```

Comment:
since hidden_layers=() is explicitly handled in this function, there should probably be a test case for it.

Reply:
good point, just added for both vjp and jvp tests.